spec/registry.dtd

```diff
@@ -1,6 +1,6 @@
-<!ELEMENT registry (function*|pattern*)>
+<!ELEMENT registry (function|pattern)*>
```

Comment:
In #420 @mihnita suggested changing this to:

```dtd
<!ELEMENT registry (function,pattern)*>
```

...but IIUC, this requires `function` and `pattern` to appear one after the other, in this specific order, i.e. interleaved. Instead, in this PR I'm changing the original `(function*|pattern*)` — which wrongly meant _any number of `function` OR any number of `pattern`, but never both_ — to `(function|pattern)*`.